src/Streamly/Streams/StreamD.hs

```diff
@@ -2587,10 +2606,11 @@ decodeUtf8 (Stream step state) = Stream step' (state, FreshPoint)
         return $
             case r of
                 Yield c1 s
-                    | c1 < 0x80 || c1 >= 0xc0 -> error "Invalid Sequence"
-                    | otherwise -> Skip (s, YieldPoint $ chr2 c0 c1)
+                    | c1 < 0x80 || c1 >= 0xc0 ->
+                        transliterateOrError "Invalid Sequence" s
+                    | otherwise -> Skip (s, YieldAndContinue $ chr2 c0 c1)
                 Skip s -> Skip (s, Two c0)
-                Stop -> error "Input Underflow"
+                Stop -> inputUnderflow st
```

Comment:
Here we can say something like "need one more codepoint after 0xc0" to know the context?